notmuch/config.c

```diff
@@ -30,14 +30,40 @@
 #include <stddef.h>
 #include <config/lib.h>
 #include <stdbool.h>
+#include "notmuch/private.h"
+
+/**
+ * is_valid_notmuch_url - Checks that a URL is in required form.
+ * @retval true url in form notmuch://<absolute path>
+ * @retval false url is not in required form.
+ */
+static bool is_valid_notmuch_url(const char *uri)
+{
+  return mutt_istr_startswith(uri, NmUrlProtocol) && uri[NmUrlProtocolLen] == '/';
+}
+
+/**
+ * nm_default_url_validator - Ensure nm_default_url is in form notmuch://<absolute path> - Implements ConfigDef::validator()
+ */
+static int nm_default_url_validator(const struct ConfigSet *cs, const struct ConfigDef *cdef, intptr_t value, struct Buffer *err)
+{
+  const char *uri = (const char *) value;
+  if (!is_valid_notmuch_url(uri))
+  {
+    mutt_buffer_printf(err, _("nm_default_url must be in form: notmuch://<absolute path> . Current value %s"), uri);
```

Comment:
It'd be nice to show the user the current value, but mind if we shorten it to just the hint?

On an 80 char screen, you're likely to see:
```
|                                                                                |
|nm_default_url must be in form: notmuch://<absolute path> . Current value notmuc|
|                                                                                |
```

Reply:
Hmm...  That looks nice, but...

I wasn't thinking about config files, just the on-screen look.
Unfortunately, I've just tested that with newlines and it doesn't work.

There is a hack `mutt_window_reflow_message_rows()`, but intercepting the message would be a pain.

Reply:
Ah, I see what you mean. When a user runs `:set nm_default_url`, right? I'll trim up the message.

Reply:
thanks

Reply:
Trimmed the message a bit to reduce character count by 13.

`set nm_default_url="notmuch://"` now returns:

 ```
nm_default_url must be: notmuch://<absolute path> . Current: notmuch://
```

from:

```
nm_default_url must be in form: notmuch://<absolute path> . Current value notmuch://
```